lib/services/login.dart

```diff
@@ -0,0 +1,32 @@
+import 'dart:async';
+import 'dart:convert';
+import 'package:shared_preferences/shared_preferences.dart';
+import 'post.dart';
+import '../model/user.dart';
+
+class Login {
+  static final baseUrl = "https://apistaginghr.excellencetechnologies.in";
+  static final loginUrl = baseUrl + "/attendance/API_HR/api.php";
+  static final apiKey = null;
+  Post _post = Post();
+  Future<User> login(String username, String password) {
+    //print(username + "" + password);
+    Map data = {
+      "token": apiKey,
+      "action": "login",
+      "username": username,
+      "password": password
+    };
+    return _post
+        .post(loginUrl, body: json.encode(data))
+        .then((dynamic res) async {
+      if (res["error"] >= 1) throw new Exception(res["data"]["message"]);
+      if (res["error"] == 0) {
+        SharedPreferences prefs = await SharedPreferences.getInstance();
```

Comment:
all local stroage stuff should be moved to a seperate service like localstorage.dart

Reply:
ok creating storage service